### ISSUE TITLE
Clarification on Inference Model selection and Training/Validation logic

### ISSUE THREAD

**Issue by alishan2040** [evenchen6/CAN_GAN_Anomaly#5]
Hi @evenevennnn, @evenchen6 , I am using this repository and noticed some ambiguity regarding the model loading process during inference. I would appreciate clarification on the following points:
**1. Inference defaults to Epoch 0** In model/model.py, the validate() function calls self.load_pkl() without arguments. Since the load_pkl method defaults to epoch=0, the inference phase currently evaluates the weights from the very first epoch rather than the best-performing one:
`
   def validate(self):
    
    # ...
    
    self.load_pkl() # Defaults to epoch 0
    
    self.val_openset(0)
`

Is this intended, or should there be a mechanism to automatically load the model with the best validation metrics?

**2. Identifying the "Best" Model** The current train() loop saves a .pkl file for every epoch but does not track or flag a "best" model (e.g., best_discriminator.pkl).

- What is the recommended way to select the optimal epoch for the inference mode?

**3. Evaluation Strategy (Previous vs. Current)** In a previous version/workflow, I evaluated every saved model from every epoch on the test data and kept the highest result. However, I am concerned about data leakage/overfitting to the test set using this approach. Was the intended workflow to use a separate validation split to select the best epoch before final testing?

**4. Data Leakage in Validation** The training loop calls self.val_openset() using self.dataloader.valid. In many setups, this valid set contains the actual test data (Unknowns). If we select the "best" model based on test performance, it introduces data leakage. Was the intended workflow to use a separate validation split of the "Known" data for epoch selection?
**5. Training Protocol: Fixed Knowns vs. Retraining Loop** The `run_can.sh` script suggests a loop where the model is retrained for every abnormal class. My use case involves a fixed set of knowns ($C_1, C_2, C_3$) and multiple separate unknowns ($U_1, U_2, U_3$). Since the training distribution ($C_1, C_2, C_3$) is constant, is it correct to assume I only need to train once and then run inference against $U_1, U_2,$ and $U_3$? Or is there a specific adversarial reason why the model needs to be "aware" of which class is designated as abnormal during the training phase via the --abnormal_class flag?

Thank you for your help and for sharing this work!
